### PR TITLE
chore: create appsmith schema for the pg branch with user and password

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description
Create schema for the pg used by Appsmith. The default public schema can cause issues and creating a new schema allows us to have isolation with the other data in pg. This also creates a user and password which is used to access the schema. 

## Automation

/ok-to-test tags="@tag.Sanity"

### :mag: Cypress test results

> [!CAUTION]
> 🔴 🔴 🔴 Some tests have failed.
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11051541656>
> Commit: 973dabfc3946157a70210e5858d5bceac2a6e293
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11051541656&attempt=1&selectiontype=test&testsstatus=failed&specsstatus=fail" target="_blank">Cypress dashboard</a>.
> Tags: @tag.Sanity
> Spec: 
> The following are new failures, please fix them before merging the PR: <ol>
> <li>cypress/e2e/Regression/ClientSide/Git/GitSync/SwitchBranches_spec.js</ol>
> <a href="https://internal.appsmith.com/app/cypress-dashboard/identified-flaky-tests-65890b3c81d7400d08fa9ee3?branch=master" target="_blank">List of identified flaky tests</a>.
> <hr>Thu, 26 Sep 2024 12:13:36 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Introduced automatic initialization of the PostgreSQL database schema for improved setup.
	- Enhanced handling of database credentials with defaults for user and password.

- **Bug Fixes**
	- Improved logging for schema creation success or failure.

### CODE REVIEW

File: deploy/docker/fs/opt/appsmith/entrypoint.sh [hunk 1]
```diff
@@ -435,6 +435,51 @@ init_postgres() {
     runEmbeddedPostgres=0
   fi
 
+  # Create the appsmith schema
+  init_postgres() {
```
Comment (abhvsn) [inline deploy/docker/fs/opt/appsmith/entrypoint.sh:439]: We already have the init_postgres method in this file 🤔?

File: deploy/docker/fs/opt/appsmith/entrypoint.sh [hunk 2]
```diff
@@ -435,6 +435,51 @@ init_postgres() {
...
+    echo "Initializing PostgreSQL..."
+
+    # Check if APPSMITH_DB_URL is a PostgreSQL URL
+    if [[ -n "$APPSMITH_DB_URL" && "$APPSMITH_DB_URL" == postgres*://* ]]; then
```
Comment (abhvsn) [inline deploy/docker/fs/opt/appsmith/entrypoint.sh:443]: As of now the entrupoint script which is responsible for creating an env file is starting with MongoDB url. How did we tested this change?

File: deploy/docker/fs/opt/appsmith/entrypoint.sh [hunk 3]
```diff
@@ -435,6 +435,51 @@ init_postgres() {
...
+
+      # Set defaults for host, port, and database name if not set
+      DB_HOST=${DB_HOST:-localhost}
+      DB_PORT=${DB_PORT:-5432}
+      DB_NAME=${DB_NAME:-appsmith}
+
+      echo "Connecting to PostgreSQL at $DB_HOST:$DB_PORT with user $DB_USER"
+
+      # Execute SQL to create the appsmith schema if it doesn't exist (single line)
+      PGPASSWORD=$DB_PASSWORD psql -h $DB_HOST -p $DB_PORT -U $DB_USER -d $DB_NAME -c "CREATE SCHEMA IF NOT EXISTS appsmith;"
```
Comment (abhvsn) [inline deploy/docker/fs/opt/appsmith/entrypoint.sh:468]: Should we also create a separate user in case of embedded Postgres DB alongwith the password.

**Review by abhvsn — CHANGES_REQUESTED** [review]
Added few comments, but I'm of the opinion if you make these changes on pg branch and then port it here it will be easier to test these changes.

File: deploy/docker/fs/opt/appsmith/entrypoint.sh (hunk 3 above)
Comment (AnaghHegde) [reply deploy/docker/fs/opt/appsmith/entrypoint.sh:468]: What do you mean by separate user for embeded db? The user is hardcoded to appsmith when the username or password is empty

File: deploy/docker/fs/opt/appsmith/entrypoint.sh (hunk 2 above)
Comment (AnaghHegde) [reply deploy/docker/fs/opt/appsmith/entrypoint.sh]: I am testing this on the pg branch.

File: deploy/docker/fs/opt/appsmith/entrypoint.sh (hunk 3 above)
Comment (abhvsn) [reply deploy/docker/fs/opt/appsmith/entrypoint.sh]: I was referring to create user statement, you can find more details https://www.postgresql.org/docs/8.0/sql-createuser.html

**Pull request closed without merging**